lib/retrieval.js

```diff
@@ -0,0 +1,48 @@
+import fetch from 'node-fetch'
```

Comment:
There's no need for using `node-fetch` as `fetch` is part of the standard library for node versions 18+ (we're using 22).